sda-commons-server-key-mgmt/src/main/java/org/sdase/commons/keymgmt/manager/MapOrFailKeyMapper.java

```diff
@@ -2,21 +2,60 @@
 
 import java.util.Locale;
 import java.util.Objects;
+import java.util.Optional;
 import org.sdase.commons.keymgmt.model.KeyMappingModel;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 public class MapOrFailKeyMapper implements KeyMapper {
 
+  private static final Logger LOG = LoggerFactory.getLogger(MapOrFailKeyMapper.class);
+
+  private static final String PLACEHOLDER = "UNKNOWN_KEY_VALUE";
+
   private final KeyMappingModel mappingModel;
 
-  public MapOrFailKeyMapper(KeyMappingModel mappingModel) {
+  private final boolean useApiToImplPlaceholder;
+
+  private final boolean useImplToApiPlaceholder;
+
+  /**
+   * Creates a new {@link MapOrFailKeyMapper}, optionally using placeholders for missing mappings.
+   *
+   * @param mappingModel the mapping model to use
+   * @param useApiToImplPlaceholder if true, a placeholder is used for missing mappings from API to
+   *     implementation
+   * @param useImplToApiPlaceholder if true, a placeholder is used for missing mappings from
+   *     implementation to API
+   * @return a new {@link MapOrFailKeyMapper}
+   */
+  public MapOrFailKeyMapper(
+      KeyMappingModel mappingModel,
+      boolean useApiToImplPlaceholder,
+      boolean useImplToApiPlaceholder) {
     this.mappingModel = mappingModel;
+    this.useApiToImplPlaceholder = useApiToImplPlaceholder;
+    this.useImplToApiPlaceholder = useImplToApiPlaceholder;
   }
 
   @Override
   public String toImpl(String value) {
     return mappingModel
         .getMapping()
         .mapToImpl(value)
+        .or(
+            () -> {
+              if (useApiToImplPlaceholder) {
+                LOG.warn(
```

Comment:
Is warn level appropriate in case a service is explicitly configured to use default values? Warn level logs might flood not only the log but also monitoring or alerting systems. But info level logs can still be picked selectively to get this information if needed to clean up data or fix the system.
If it is required, I would appreciate to read in the docs, that unknown values are still considered as mistake in the data and result in a warn log.